File: common/src/main/clojure/com/github/lukebemish/clojurewrapper/api/sided.clj
(ns com.github.lukebemish.clojurewrapper.api.sided
  (:require [com.github.lukebemish.clojurewrapper.api.platform :as platform])
  (:import (com.github.lukebemish.clojurewrapper.env Sided)))

(defn client? [] (Sided/isClient))
(defn server? [] (Sided/isServer))

(def run-sided platform/run-platform)

File: common/src/main/java/com/github/lukebemish/clojurewrapper/env/Sided.java
package com.github.lukebemish.clojurewrapper.env;

import dev.architectury.injectables.annotations.ExpectPlatform;

public class Sided {
    @ExpectPlatform
    public static boolean isServer() {
        throw new AssertionError();
    }

    @ExpectPlatform
    public static boolean isClient() {
        throw new AssertionError();
    }
}
